guides/advanced_guides/search_parameters.md

```diff
@@ -1,71 +1,146 @@
+| **[attributesToHighlight](/guides/advanced_guides/search_parameters.md#attributes-to-highlight)** | Attributes whose values will contain highlighted matching terms |    `none`     |
+| **[filters](/guides/advanced_guides/search_parameters.md#filters)**                               | Filter queries by an attribute value                            |    `none`     |
+| **[matches](/guides/advanced_guides/search_parameters.md#matches)**                               | Defines whether the raw matches should be returned or not       |    `false`    |
 
 ## Query (q)
 
-The query parameter is the **only mandatory** parameter.
+The query parameter is the **only mandatory** parameter. This is the string used by the search engine to find relevant documents.
 
-It is the string the search engine uses to find relevant documents.
+`q=<String>`
+
+- `<String>` (Mandatory, string)
+
+  The query string.
+
+#### Example
+
+Suppose you would like to search `shifu` in a movie database, you would send:
+
+```bash
+$ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
+      -d q=shifu
+```
 
 ::: tip
-Although the API will send back documents even with only one letter, the more precise the search query is, the faster the API responds.
+Although the API will send back documents even if the query is only one character long, the more precise the search query is, the faster the API will respond.
 :::
 
 ## Offset
 
-`offset=<Integer>`.
+A number of **documents to skip**.
+
+`offset=<Integer>`
 
-X first documents to skip. This is helpful for **pagination**
+- `<Integer>` (Optional, positive integer, defaults to `0`)
+
+  If the value of the parameter `offset` is _n_, _n_ first documents to skip. This is helpful for **pagination**.
+
+#### Example
+
+If you want to skip the **first** document, set `offset` to `1`.
+
+```bash
+$ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
+      -d q=shifu \
+      -d offset=1
+```
 
 ## Limit
 
-`limit=<Integer>`.
+Set a **limit to the number of documents returned** by search queries.
+
+`limit=<Integer>`
+
+- `<Integer>` (Optional, positive integer, defaults to `20`)
 
-X number of documents in the search query response. This is helpful for **pagination**
+  If the value of the parameter `limit` is _n_, there will be _n_ documents in the search query response. This is helpful for **pagination**.
+
+#### Example
+
+If you want to get only **two** documents, set `limit` to `2`.
+
+```bash
+$ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
+      -d q=shifu \
+      -d limit=2
+```
 
 ## Attributes to retrieve
 
-`attributesToRetrieve=<Attribute>,<Attribute>,...`.
+Attributes to **display** in the returned documents.
+
+`attributesToRetrieve=<Attribute>,<Attribute>,...`
 
-Attributes that will appear in the returned documents.
+- `<Attribute>` (Optional, string, Defaults to `*`)
+
+  Comma-separated list of attributes that shall be in the returned documents.
```

Comment:
```suggestion
  Comma-separated list of attributes whose fields will be present in the returned documents.
```

Reply:
Sorry, it was outdated as I replaced `shall`. I applied your changes.